internal/config/config.go

```diff
@@ -0,0 +1,177 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+)
+
+const (
+	hcloudToken    = "HCLOUD_TOKEN"
+	hcloudEndpoint = "HCLOUD_ENDPOINT"
+	hcloudNetwork  = "HCLOUD_NETWORK"
+	hcloudDebug    = "HCLOUD_DEBUG"
+
+	hcloudInstancesAddressFamily = "HCLOUD_INSTANCES_ADDRESS_FAMILY"
+
+	// Disable the "master/server is attached to the network" check against the metadata service.
+	hcloudNetworkDisableAttachedCheck = "HCLOUD_NETWORK_DISABLE_ATTACHED_CHECK"
+	hcloudNetworkRoutesEnabled        = "HCLOUD_NETWORK_ROUTES_ENABLED"
+
+	hcloudLoadBalancersEnabled               = "HCLOUD_LOAD_BALANCERS_ENABLED"
+	hcloudLoadBalancersLocation              = "HCLOUD_LOAD_BALANCERS_LOCATION"
+	hcloudLoadBalancersNetworkZone           = "HCLOUD_LOAD_BALANCERS_NETWORK_ZONE"
+	hcloudLoadBalancersDisablePrivateIngress = "HCLOUD_LOAD_BALANCERS_DISABLE_PRIVATE_INGRESS"
+	hcloudLoadBalancersUsePrivateIP          = "HCLOUD_LOAD_BALANCERS_USE_PRIVATE_IP"
+	hcloudLoadBalancersDisableIPv6           = "HCLOUD_LOAD_BALANCERS_DISABLE_IPV6"
+
+	hcloudMetricsEnabled = "HCLOUD_METRICS_ENABLED"
+	hcloudMetricsAddress = ":8233"
+)
+
+type HCloudClientConfiguration struct {
+	Token    string
+	Endpoint string
+	Debug    bool
+}
+
+type MetricsConfiguration struct {
+	Enabled bool
+	Address string
+}
+
+type AddressFamily string
+
+const (
+	AddressFamilyDualStack AddressFamily = "dualstack"
+	AddressFamilyIPv6      AddressFamily = "ipv6"
+	AddressFamilyIPv4      AddressFamily = "ipv4"
+)
+
+type InstanceConfiguration struct {
+	AddressFamily AddressFamily
+}
+
+type LoadBalancerConfiguration struct {
+	Enabled               bool
+	Location              string
+	NetworkZone           string
+	DisablePrivateIngress bool
+	UsePrivateIP          bool
+	DisableIPv6           bool
+}
+
+type NetworkConfiguration struct {
+	NameOrID             string
+	DisableAttachedCheck bool
+}
+
+type RouteConfiguration struct {
+	Enabled bool
+}
+
+type HCCMConfiguration struct {
+	HCloudClient HCloudClientConfiguration
+	Metrics      MetricsConfiguration
+	Instance     InstanceConfiguration
+	LoadBalancer LoadBalancerConfiguration
+	Network      NetworkConfiguration
+	Route        RouteConfiguration
+}
+
+func Read() (HCCMConfiguration, error) {
+	var err error
+	// Collect all errors and return them as one.
+	// This helps users because they will see all the errors at once
+	// instead of having to fix them one by one.
+	var errs []error
+	var config HCCMConfiguration
+
+	config.HCloudClient.Token = os.Getenv(hcloudToken)
+	config.HCloudClient.Endpoint = os.Getenv(hcloudEndpoint)
+	config.HCloudClient.Debug, err = getEnvBool(hcloudDebug)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	config.Metrics.Enabled = os.Getenv(hcloudMetricsEnabled) != "false"
```

Comment:
Why not use `getEnvBool()` ?

Reply:
This (and the other two codepaths you commented) was done for two reasons:
- **Backwards compatibility**: We previously interpreted everything but `"false"` as `true`, `getEnvBool()` has other rules
- **Default**: `getEnvBool()` defaults to `false`, while we want these fields to default to `true`.

I will add a comment to the function to explain this.

Reply:
Updated to your suggestion, will also add a note in the release notes for this.